src/hooks/useSelectedRevisions.ts

```diff
@@ -23,22 +25,67 @@ const useSelectRevision = () => {
   const addSelectedRevisions = () => {
     const newSelected = [...selectedRevisions];
     newSelected.push(...baseCheckedRevisions, ...newCheckedRevisions);
-    //remove duplicates of the same revision
-    const filteredSelected = newSelected.filter(
-      (revision, index, self) =>
-        self.findIndex((r) => r.id === revision.id) === index,
-    );
+    //create a new array with unique values
+    const filteredSelected = [...new Set(newSelected)];
 
     dispatch(setSelectedRevisions({ selectedRevisions: filteredSelected }));
```

Comment:
thought for a later patch (but not for now): using a Set for `selectedRevisions` would make it unnecessary to convert it back and forth between arrays. But we would till have to create a new set when changing it, and this may have performance concerns.
although in this case where it's at most 3 revisions the performance concerns are not an issue.